### PR TITLE
Accordion mode variant rename

### PR DESCRIPTION
**Pull request by EmelieLitwin**
**Describe pull-request**  
Updating mode variant naming

**Solving issue**  
_In case of GitHub issue add # plus the number of the issue (for example #123) OR if it is Azure then AB# and number of ticket_
Fixes: [AB#2970](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/2970)

### CODE REVIEW

**Comment by charlottelorensson** [comment]
The primary version (on white) has grey-50 as preview background in Storybook so the hover state for primary is not visible (grey-50 on a grey-50 background). The secondary works cause then the background should be grey-50.

**Comment by EmelieLitwin** [comment]
> The primary version (on white) has grey-50 as preview background in Storybook so the hover state for primary is not visible (grey-50 on a grey-50 background). The secondary works cause then the background should be grey-50.
> 
> Although, the issue will be reversed if resolved by changing the background to white I guess. Could the background color change as well when switching between primary and secondary?

Yes, there is nothing wrong with the component but I also wish we could make it så the background changed when the user switched mode variant. But I'm not sure how to do this so hat it works with both modes. I can create a ticket for that. There is already a ticket for changing background color I think.

File: tegel/src/components/accordion/accordion-theme-vars.scss [hunk 1]
```diff
@@ -2,31 +2,43 @@
...
   --sdds-accordion-border-focus: var(--sdds-grey-500);
   --sdds-accordion-color-disabled: var(--sdds-grey-800);
 
-  .sdds-on-white-background {
-    --sdds-accordion-background-hover: var(--sdds-accordion-background-hover-on-white);
+   .sdds-mode-variant-primary {
+    --sdds-accordion-background-hover: var(--sdds-accordion-background-hover-primary);
+  }
+  .sdds-mode-variant-secondary {
+    --sdds-accordion-background-hover: var(--sdds-accordion-background-hover-secondary);
```
Comment (mJarsater) [inline tegel/src/components/accordion/accordion-theme-vars.scss:42]: I think this change (and I guess it goes for all made-variant changes on the native components) need to be added to the migration docs.

Comment (EmelieLitwin) [reply tegel/src/components/accordion/accordion-theme-vars.scss:42]: > I think this change (and I guess it goes for all made-variant changes on the native components) need to be added to the migration docs.

Good point! I think I added it for button WC and native but not the others.

Comment (mJarsater) [reply tegel/src/components/accordion/accordion-theme-vars.scss:42]: Nice! I guess it's really the most important for the native components. Maybe we should create a ticket for that?

**Comment by EmelieLitwin** [comment]
@mJarsater Updated mode variant prop to null. What do you think?

**Review by mJarsater — APPROVED** [review]
LTGM 🎸

**Pull request merged**